include/ddc/kernels/splines/splines_linear_problem_band.hpp

```diff
@@ -0,0 +1,171 @@
+        assert(m_kl <= mat_size);
+        assert(m_ku <= mat_size);
+
+        /*
+         * Given the linear system A*x=b, we assume that A is a square (n by n)
+         * with ku super-diagonals and kl sub-diagonals.
+         * All non-zero elements of A are stored in the rectangular matrix q, using
+         * the format required by DGBTRF (LAPACK): diagonals of A are rows of q.
+         * q has 2*kl rows for the subdiagonals, 1 row for the diagonal, and ku rows
+         * for the superdiagonals. (The kl additional rows are needed for pivoting.)
+         */
+        Kokkos::deep_copy(m_q, 0.);
+    }
+
+    virtual double get_element(std::size_t const i, std::size_t const j) const override
+    {
+        assert(i < size());
+        assert(j < size());
+        /*
+         * The "row index" of the band format storage identify the (sub/supra)-diagonal
+         * while the column index is actually the column index of the matrix. Two layouts
+         * are supported by LAPACKE. The m_kl first lines are irrelevant for the storage of 
+         * the matrix itself but required for the storage of its LU factorization.
+         */
+        if (i >= std::max((std::ptrdiff_t)0, (std::ptrdiff_t)j - (std::ptrdiff_t)m_ku)
+            && i < std::min(size(), j + m_kl + 1)) {
+            // return m_q(j * m_c + m_kl + m_ku + i - j); //LAPACK_COL_MAJOR
+            return m_q((m_kl + m_ku + i - j) * size() + j); //LAPACK_ROW_MAJOR
+        } else {
+            return 0.0;
+        }
+    }
+
+    virtual void set_element(std::size_t const i, std::size_t const j, double const aij) override
+    {
+        assert(i < size());
+        assert(j < size());
+        /*
+         * The "row index" of the band format storage identify the (sub/supra)-diagonal
+         * while the column index is actually the column index of the matrix. Two layouts
+         * are supported by LAPACKE. The m_kl first lines are irrelevant for the storage of 
+         * the matrix itself but required for the storage of its LU factorization.
+         */
+        if (i >= std::max((std::ptrdiff_t)0, (std::ptrdiff_t)j - (std::ptrdiff_t)m_ku)
+            && i < std::min(size(), j + m_kl + 1)) {
+            // m_q(j * m_c + m_kl + m_ku + i - j) = aij; // LAPACK_COL_MAJOR
+            m_q((m_kl + m_ku + i - j) * size() + j) = aij; // LAPACK_ROW_MAJOR
+        } else {
+            assert(std::fabs(aij) < 1e-20);
+        }
+    }
+
+    /**
+     * @brief Perform a pre-process operation on the solver. Must be called after filling the matrix.
+     *
+     * LU-factorize the matrix A and store the pivots using the LAPACK dgbtrf() implementation.
+     */
+    void setup_solver() override
+    {
+        int const info = LAPACKE_dgbtrf(
+                LAPACK_ROW_MAJOR,
+                size(),
+                size(),
+                m_kl,
+                m_ku,
+                m_q.data(),
+                size(), // size() if LAPACK_ROW_MAJOR, m_c if LAPACK_COL_MAJOR
```

Comment:
Can you provide a version using a 2D Kokkos View ? this parameter would then become m_q.stride(0) whatever the layout.

Reply:
Yes that's true

Reply:
Actually that's not, it has to be m_q.stride(0) if ROW_MAJOR and m_q.stride(1) if COL_MAJOR. But the 2D Kokkos::View makes sense for sure.